pct.sh

```diff
@@ -18,23 +18,99 @@ if [[ -n ${EXTRA_MAVEN_PROPERTIES-} ]]; then
 fi
 
 #
-# Plugin Compatibility Tester (PCT) requires custom --add-opens directives when running on Java 17.
-# As a temporary workaround, we pass in these directives when invoking PCT. When
-# jenkinsci/plugin-compat-tester#352 is merged and released, and when this repository has upgraded
-# to that release, this workaround can be deleted.
+# Grab the Jenkins version from the WAR file so that we can pass it in via jenkins.version. This is
+# needed because HPI Plugin requires the version of the WAR passed in via overrideWar to be
+# identical to jenkins.version. If we do not explicitly pass in jenkins.version, then the
+# jenkins.version defined in the plugin's pom.xml file will be used, which may not match the version
+# of the WAR under test.
 #
+mkdir pct-work
+pushd pct-work
+jar xf ../megawar.war META-INF/MANIFEST.MF
+JENKINS_VERSION=$(grep Jenkins-Version META-INF/MANIFEST.MF | awk '{print $2}')
+popd
+rm -rf pct-work
+MAVEN_PROPERTIES+=":jenkins.version=${JENKINS_VERSION}:overrideWar=$(pwd)/megawar.war:useUpperBounds=true"
+
+#
+# The overrideWar option is available in HPI Plugin 3.29 or later, but many plugins under test
+# still use an older plugin parent POM and therefore an older HPI plugin version. As a temporary
+# workaround, we override the HPI plugin version to the latest version.
+#
+# TODO When all plugins in the managed set are using a plugin parent POM with HPI Plugin 3.29 or
+# later (i.e., plugin parent POM 4.44 or later), this can be deleted.
```

Comment:
Taking on a medium amount of technical debt here, which I felt was the more reasonable alternative to updating the plugin POM and spinning releases for 100 or so plugins, a massive exercise. I plan to keep this up-to-date as new versions of HPI plugin are released. Hopefully in a year or so the vast majority of plugins will have been updated over the course of routine work, at which point we can sweep through the stragglers and eliminate this technical debt.